runtime/gc_modron_startup/mminit.cpp

```diff
@@ -3054,7 +3054,23 @@ gcReinitializeDefaultsForRestore(J9VMThread* vmThread)
 	bool result = true;
 
 	PORT_ACCESS_FROM_JAVAVM(vm);
-
+	OMRPORT_ACCESS_FROM_J9PORT(PORTLIB);
+	/* Note here we update this parameter which represents the machine physical memory,
+	 * but not the original heap geometry from the snapshot run. We just force the heap
+	 * not to go beyond a certain thereshold calculated based on the new (restore) memory
+	 * availalibity through setting SoftMx.
+	 */
+	extensions->usablePhysicalMemory = omrsysinfo_get_addressable_physical_memory();
+	/* We use false here for computeDefaultMaxHeapForJava(), since the restore
+	 * path is only active for releases after Java 8 releases
+	 */
+	uintptr_t candidateSoftMx = extensions->computeDefaultMaxHeapForJava(false);
+	/* if snapshot machine's memory is higher than the restore machine */
```

Comment:
you can generalize this comment, something like: we will set softMx value only if maxHeap calculation returned us a smaller value than existing maxHeap or softMx values as inherited from/established at snapshot run